src/ape_ethereum/transactions.py

```diff
@@ -262,7 +262,11 @@ def decode_logs(
             Union[List[Union[EventABI, "ContractEvent"]], Union[EventABI, "ContractEvent"]]
         ] = None,
     ) -> ContractLogContainer:
-        if abi is not None:
+        if not self.logs:
```

Comment:
author's note: the changes in the file are nice, but are a product of investigating missing events on a receipt.

so the receipt was actually failing super early, and thus didnt have the logs, so no actual change here but 1 small perf and some refactoring.